bot/exts/evergreen/latex.py

```diff
@@ -0,0 +1,57 @@
+from io import BytesIO
+from typing import Union
+
+import discord
+import matplotlib.pyplot as plt
+from discord.ext import commands
+
+# configure fonts and colors for matplotlib
+plt.rcParams.update(
+    {
+        "font.size": 16,
+        "mathtext.fontset": "cm",  # Computer Modern font set
+        "mathtext.rm": "serif",
+        "figure.facecolor": "38383F",  # matches Discord's dark mode background color
+        "text.color": "white",
+    }
+)
+
+
+class Latex(commands.Cog):
+    """Renders latex."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @staticmethod
+    def _render(text: str) -> Union[BytesIO, str]:
+        """Return the rendered image if latex compiles without errors, otherwise return the error message."""
+        text = text.replace(r"\\", "$\n$")  # matplotlib uses \n for newlines, not \\
+        fig = plt.figure()
+
+        try:
+            fig.text(0, 1, text, horizontalalignment="left", verticalalignment="top")
+
+            rendered_image = BytesIO()
+            plt.savefig(rendered_image, bbox_inches="tight", dpi=600)
+            rendered_image.seek(0)
+            return rendered_image
+
+        except ValueError as e:
+            return str(e)
+
+    @commands.command()
+    async def latex(self, ctx: commands.Context, *, text: str) -> None:
+        """Renders the text in latex and sends the image."""
+        async with ctx.typing():
+            image = self._render(text)
+
+            if isinstance(image, BytesIO):
+                await ctx.send(file=discord.File(image, "latex.png"))
+            else:
+                await ctx.send(image)
```

Comment:
Thanks for the PR! Just a small suggestion:
I think instead of returning a `Union[BytesIO, str]` after a `try-except`, it'd be better if we catch that error within `latex` instead. You could raise a `commands.BadArgument` instead of sending the exception message directly as well. So possibly something like:
```py
    @commands.command()
    async def latex(self, ctx: commands.Context, *, text: str) -> None:
        """Renders the text in latex and sends the image."""
        async with ctx.typing():
            try:
                image = self._render(text)
                await ctx.send(file=discord.File(image, "latex.png"))
            except ValueError as e:
                raise commands.BadArgument("Failed to render.")
```

Reply:
Alright, once I do that would I have to change anything in the `error_handler.py`?
Also, I think if the rendering fails, the users might want to see where the syntax error is in their latex

Reply:
Not entirely sure on the former, but assuming `e` has the information for that, you could interpolate it

Reply:
https://github.com/python-discord/sir-lancebot/blob/964112352809bcec5b13fe5da93b15b4ade1b196/bot/exts/evergreen/error_handler.py#L94 catches BadArguments. The string that you pass when raising will be included in the output, so you could include what you want to output in that there.